odl/operator/fn_ops.py

```diff
@@ -0,0 +1,454 @@
+        >>> x = X.element(range(X.size))
+        >>> A = odl.VectoriseOperator(X)
+        >>> A(x)
+        rn(12).element([0.0, 1.0, 2.0, ..., 9.0, 10.0, 11.0])
+        >>> A(x)  # Out-of-place
+        rn(12).element([0.0, 1.0, 2.0, ..., 9.0, 10.0, 11.0])
+        """
+        if not isinstance(domain, (LinearSpace, Field)):
+            raise TypeError('`space` {!r} not a `LinearSpace` or `Field` '
+                            'instance'.format(domain))
+        
+        self.__scaling = scaling        
+        range = fn(domain.size, dtype=domain.dtype)
+        super().__init__(domain, range, linear=True)
+
+    def _call(self, x, out=None):
+        """Collect indices."""
+        if out is None:
+            out = self.scaling * np.ravel(x)
+        else:
+            out[:] = self.scaling * np.ravel(x)
+        return out
+
+    @property
+    def scaling(self):
+        """scaling factor."""
+        return self.__scaling
+    
+    @property
+    def adjoint(self):
+        """Adjoint of the vectorising is creating an element with these components
+        with an appropriate weighting.
+
+        Returns
+        -------
+        adjoint : `EmbeddingOperator`
+
+        Examples
+        --------
+        >>> X = odl.uniform_discr(min_pt=[-1, -1], max_pt=[1, 1], shape=[2, 3])
+        >>> A = odl.VectoriseOperator(X)
+        >>> x = A.domain.element(range(A.domain.size))
+        >>> A.adjoint(A(x)).inner(x) - A(x).inner(A(x)) < 1e-10
+        True
+        >>> A.adjoint(A(x)).inner(x) - A(x).inner(A(x)) < 1e-10 # Out-of-place
+        True
+        """
+        c = getattr(self.domain, 'cell_volume', 1.0) # TODO: This only works for a subset of function spaces
+        return EmbeddingOperator(self.domain, self.scaling/c)
+
+    @property
+    def inverse(self):
+        """Inverse of the vectorising is creating an element with these components.
+
+        Returns
+        -------
+        adjoint : `EmbeddingOperator`
+
+        Examples
+        --------
+        >>> X = odl.uniform_discr(min_pt=[-1, -1], max_pt=[1, 1], shape=[3, 4])
+        >>> x = X.element(range(X.size))
+        >>> A = odl.VectoriseOperator(X)
+        >>> (A.inverse(A(x)) - x).norm() < 1e-10
+        True
+        >>> (A.inverse(A(x)) - x).norm() < 1e-10 # Out-of-place
+        True
+        """
+        return EmbeddingOperator(self.domain, 1./self.scaling)
+
+    def __repr__(self):
+        """Return ``repr(self)``."""
+        return '{}({!r})'.format(self.__class__.__name__, self.domain)
+
+    def __str__(self):
+        """Return ``str(self)``."""
+        return repr(self)
+
+class EmbeddingOperator(Operator):
+
+    """Operator that creates an element of a vector space given its coefficients.
+    The operator is defined by:
+
+        ``EmbeddingOperator(X)(f) == X.element(f)``
+        
+    For the adjoint it is important that the element is scaled properly.
+    """
+
+    def __init__(self, range, scaling=1):
```

Comment:
Slightly weird asymmetry that you need to give the range here. No need to fix it now, but both guys should have the flat space as an optional argument so you can use a `DiscreteLp` as a flat space, for example.
Write a TODO comment.